### ISSUE TITLE
Automated synchronization of metadata between ILS and SDR

### ISSUE THREAD

**Issue by thatbudakguy** [sul-dlss/argo#3910]
This ticket was initially written about map metadata, but probably applies more broadly to any data linked between the ILS and SDR. It's possible that FOLIO might make this process easier to implement.

Currently, the workflow for remediating map metadata in the ILS is as follows:
1. Records are created in WorkFlows (either by batch or cataloged individually, either original or copy)
2. Maps are digitized and accessioned into Argo
3. Map metadata is remediated in WorkFlows after imaging is complete.

To reflect these changes in the SDR, a request must be manually initiated in Argo. Need: **Metadata synchronization that does not need a manual request to be initiated**.

Why: 
Increased communication between the platforms allows us to create a better user experience for researchers and students. Different and/or incorrect information in different locations creates a confusing user experience. If the information is not updated correctly, consequences can be either a minor inconvenience or that an item is not discoverable (either physically or digitally).  

Statements:
As catalogers, we often remediate records in WorkFlows to reflect updated information.  This might mean changes to the title, author, or even a new barcode for an item.  However, these updates are not pushed through into Argo automatically. 

See examples for the Conrad collection: 
PURL:[ https://purl.stanford.edu/gv410zx8596](https://purl.stanford.edu/gv410zx8596)
SW:[ https://searchworks.stanford.edu/view/13412312](https://searchworks.stanford.edu/view/13412312) 
 
In this example, we have copy cataloged the record from WorldCat.  Although we remediated the record to reflect the physical item in WorkFlows, these changes have not been automatically pushed through to the PURL. 

	
See example from the O’Neil collection
https://searchworks.stanford.edu/view/11883853 

In this example, the map had barcode 36105223421673 when it was digitized. However, some time later, we then changed the barcode in WorkFlows to 36105229150482.  In the interim, we have not yet gone into Argo to remediate this change.  Because it does not occur automatically, this can cause a lack of discoverability.  In Argo, you must use the old barcode, in Workflows, you  must use the new barcode.  Additionally, if you want the 856 in Workflows to reflect the new barcode, you likely need to re-release the druid from Argo.

**Comment by thatbudakguy** [comment]
Aspirational ticket that requires a lot more discussion & analysis; might also belong on a different repo. Just getting the discussion started.

**Comment by andrewjbtw** [comment]
Definitely a big discussion item. If the syncing is automated it's likely that Argo wouldn't be involved, but I'm hesitant to move this ticket anywhere else prior to further discussion. The whole process would likely span multiple repos anyway.

And I can confirm the last point in the ticket description: you'd have to re-release to update the 856 following a barcode change.

I think the full chain for update is:

- Update in workflows (Symphony/ILS)
- Argo: open object for editing
- Argo: refresh metadata (re-import from catkey)
- Argo: close object
- SDR system: updates the Purl with new metadata
- Argo: re-run release (to update the 856)

Another point in favor of automation is that the Purl shows the data from SDR and SearchWorks shows data from the MARC. So if the SDR data isn't updated, it will not be consistent with the MARC. All other apps that use the same metadata that the Purl displays will also be missing the update.

**Comment by andrewjbtw** [comment]
Adding a post-Symphony update: during the SDR-Folio integration work (spring 2023), there was some investigation into whether it would be possible to detect changes to ILS records so that SDR could update its records accordingly. Since that was a stretch goal, I'm not sure if the investigation could be considered conclusive, but it did not turn up a clear path where Folio could notify SDR of a change.

**Comment by edsu** [comment]
Folio's OAI-PMH service could, in theory, allow you to regularly run a [ListRecords](https://www.openarchives.org/OAI/openarchivesprotocol.html#ListRecords) API call to fetch any records that have been updated since a particular time?

**Comment by edsu** [comment]
I asked in the Folio Slack if OAI-PMH was a way to do this, and from the responses it seemed viable. Two other possible mechanisms that were mentioned included talking to the [Instance Storage API](https://s3.amazonaws.com/foliodocs/api/mod-inventory-storage/p/instance-storage.html#instance_storage_instances_get) or listening directly to the Kafka topic: `{ENV}.{TENANT}.inventory.instance`

https://folio-project.slack.com/archives/CQ7EK52LB/p1704722572561059

**Comment by thatbudakguy** [comment]
I think one place to start would be looking at our own indexing kafka queue rather than folio's internal one, in the same way that the [`KafkaFolioReader` does](https://github.com/sul-dlss/searchworks_traject_indexer/blob/main/lib/traject/readers/kafka_folio_reader.rb). That queue gets updated by regular postgres queries to folio's database and is a reliable source of information about what actually changed about a record (or as reliable as we can get from folio).

Probably we would want this update process to be entirely separate from the indexer, since it's not producing any metadata but is instead just making API calls to dor-services-app.

**Comment by edsu** [comment]
@thatbudakguy agreed! I didn't realize we already have apps that listen to Kafka queues directly! I guess things might be more complicated with SDR since there are places (ETDs) that push to the ILS, and we wouldn't want to create feedback loops.
